controllers/perconaservermysql_controller.go

```diff
@@ -99,19 +99,90 @@ func (r *PerconaServerMySQLReconciler) Reconcile(
 		return rr, errors.Wrap(err, "get CR")
 	}
 
+	if cr.ObjectMeta.DeletionTimestamp != nil {
+		return rr, r.applyFinalizers(ctx, cr)
+	}
+
 	defer func() {
 		if err := r.reconcileCRStatus(ctx, cr); err != nil {
 			l.Error(err, "failed to update status")
 		}
 	}()
-
+	
 	if err := r.doReconcile(ctx, cr); err != nil {
 		return rr, errors.Wrap(err, "reconcile")
 	}
 
 	return rr, nil
 }
 
+func (r *PerconaServerMySQLReconciler) applyFinalizers(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
+	l := log.FromContext(ctx).WithName("Finalizer")
+
+	l.Info("Applying finalizers", "CR", cr)
+
+	var err error
+
+	finalizers := []string{}
+	for _, f := range cr.GetFinalizers() {
+		switch f {
+		case "delete-ps-pods-in-order":
```

Comment:
I think `delete-mysql-pods-in-order` is a better name @hors @spron-in

Reply:
yes, better to use `delete-mysql-pods-in-order`